onnx/defs/quantization/defs.cc

```diff
@@ -97,8 +104,21 @@ ONNX_OPERATOR_SET_SCHEMA(
             "The type of the input `y_zero_point` and the output `y`.")
         .SetDoc(QuantizeLinear_ver21_doc)
         .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
-          if (ctx.hasInput(2)) {
+          auto const zp_type = ctx.hasInput(2) ? ctx.getInputType(2) : nullptr;
+          auto const output_dtype = getAttribute(ctx, "output_dtype", TensorProto::UNDEFINED);
+          if (zp_type != nullptr) {
+            const auto zp_elem_type = getTensorElementType(*zp_type);
+            if (output_dtype != TensorProto::UNDEFINED && output_dtype != zp_elem_type) {
+              fail_type_inference(
+                  "output_dtype ",
+                  ctx.getAttribute("output_dtype"),
```

Comment:
nit: why not `output_dtype` ? Also `TensorProto_DataType_Name(output_dtype)` will print the symbolic name of the datatype. (Same for `zp_elem_type` below.)

Reply:
Ok, it looks like the CI doesn't like the use of the int type in an enum context. May need an explicit cast from the int type to the enum type.